### ISSUE TITLE
Profiling

### ISSUE THREAD

**Issue by willzfarmer** [RussBaz/enforce#18]
What are the performance considerations for users wishing to use this in a production environment?

**Comment by RussBaz** [comment]
For now - Lists. Because they are fully evaluated before running the function, their performance will remain terrible till we switch to lazy evaluation of them.

I don't think there should be any other major performance problems.

**Comment by willzfarmer** [comment]
Yeah, that definitely is true for any iterable thing (except generators). Cause right now if you have a deep nested iterable thing, it will evaluate the entire structure beforehand...

I'm planning on making a new repo for profiling this. I'll keep you updated for the results.

**Comment by RussBaz** [comment]
Sure, thanks.

**Comment by ether42** [comment]
FYI, [typeguard](https://github.com/agronholm/typeguard) is checking the `__debug__` constant, it could be nice to do the same here.
